start_vscode_stanage.sh

```diff
@@ -182,70 +193,84 @@ fi
+        echo -e "\n Error: $VSC_CPUS_PER_TASK -> Incorrect format. Please specify number of tasks per node as an integer and try again\n"
         display_help
 fi
 
-# check if VSC_NUM_CPU is <= 16
-if [ "$VSC_NUM_CPU" -gt "16" ]; then
-        echo -e "Error: $VSC_NUM_CPU -> Larger than 16. No distributed memory supported, therefore the number of CPU cores needs to be smaller or equal to 16\n"
+# check if VSC_CPUS_PER_TASK is <= 64
+if [ "$VSC_CPUS_PER_TASK" -gt "64" ]; then
+        echo -e "\n Error: $VSC_CPUS_PER_TASK -> Larger than 40. No distributed memory supported, therefore the number of cpus per task needs to be smaller or equal to 40\n"
         display_help
 fi
 
-if [ "$VSC_NUM_CPU" -gt "0" ]; then
-        echo -e "Requesting $VSC_NUM_CPU CPU cores for running the code-server"
+if [ "$VSC_CPUS_PER_TASK" -gt "0" ]; then
+        echo -e "Requesting $VSC_CPUS_PER_TASK cpus per task for running the code-server"
 fi
 
 # check number of GPUs
 
 # check if VSC_NUM_GPU an integer
 if ! [[ "$VSC_NUM_GPU" =~ ^[0-9]+$ ]]; then
-        echo -e "Error: $VSC_NUM_GPU -> Incorrect format. Please specify the number of GPU as an integer and try again\n"
+        echo -e "\n Error: $VSC_NUM_GPU -> Incorrect format. Please specify the number of GPU as an integer and try again\n"
         display_help
 fi
 
-# check if VSC_NUM_GPU is <= 7
-if [ "$VSC_NUM_GPU" -gt "7" ]; then
-        echo -e "Error: No distributed memory supported, therefore number of GPUs needs to be smaller or equal to 7\n"
+# check if VSC_NUM_GPU is <= 4
+if [ "$VSC_NUM_GPU" -gt "4" ]; then
+        echo -e "\n Error: No distributed memory supported, therefore number of GPUs needs to be smaller or equal to 4\n"
         display_help
 fi
 
 if [ "$VSC_NUM_GPU" -gt "0" ]; then
         echo -e "Requesting $VSC_NUM_GPU GPUs for running the code-server"
-        VSC_SNUM_GPU="-l gpu=$VSC_NUM_GPU"
+        VSC_SNUM_GPU="--gpus-per-node=$VSC_NUM_GPU"
 else
         VSC_SNUM_GPU=""
 fi
 
-if [ ! "$VSC_NUM_CPU" -gt "0" -a ! "$VSC_NUM_GPU" -gt "0" ]; then
-        echo -e "Error: No CPU and no GPU resources requested, terminating script"
+# check if VSC_PARTITION_ID is set
+if [ "$VSC_NUM_GPU" -gt "0" ] && ! ( [ "$VSC_PARTITION_ID" == "gpu" ] || [ "$VSC_PARTITION_ID" == "gpu-a100-tmp" ] ); then
+        echo -e "\n Error: partition incorrect. Please specify either gpu or gpu-a100-tmp"
+        display_help
+elif [ "$VSC_NUM_GPU" -gt "0" ]; then
+        echo -e "Requesting partition $VSC_PARTITION_ID"
+        VSC_SNUM_GPU="--partition=$VSC_PARTITION_ID --qos=gpu $VSC_SNUM_GPU"  
+fi
+
+if [ ! "$VSC_CPUS_PER_TASK" -gt "0" -a ! "$VSC_NUM_GPU" -gt "0" ]; then
+        echo -e "\n Error: No CPU and no GPU resources requested, terminating script"
+        display_help
+fi
+
+if [ "$VSC_NUM_GPU" -gt "0" -a ! "$VSC_CPUS_PER_TASK" -gt "0" ]; then
+        echo -e "\n Error: No CPU resource requested whilst GPU resources are requested, terminating script"
         display_help
 fi
```

Comment:
We need change here to respect the Stanage GPU partitions:

https://docs.hpc.shef.ac.uk/en/latest/stanage/GPUComputingStanage.html#gsc.tab=0

Reply:
Ah missed this one. Thanks!